spock-specs/src/test/groovy/org/spockframework/smoke/extension/RetryPendingFeatureInteropSpec.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.smoke.extension
+
+
+import spock.lang.Issue
+import spock.lang.PendingFeature
+import spock.lang.PendingFeatureIf
+import spock.lang.Retry
+import spock.lang.Specification
+
+import static spock.lang.Retry.Mode.ITERATION
+import static spock.lang.Retry.Mode.SETUP_FEATURE_CLEANUP
+
+/**
+ * This specification shall test the interoperation between {@link PendingFeature} and the {@link Retry} extension.
+ */
+@Issue("https://github.com/spockframework/spock/issues/1863")
+class RetryPendingFeatureInteropSpec extends Specification {
```

Comment:
Shouldn't these tests be embedded spec tests so that you verify the actual outcome and counts?

Reply:
Isn't it also an important detail that you stop retrying as soon as the respective exception is seen and not still do all retries and just then behave differently? And whether something is successful or skipped, which both would be not failure?